### PR TITLE
Fix empty legacy meta being persisted

### PR DESCRIPTION
**Pull request by rubenwardy**

### CODE REVIEW

File: src/itemstackmetadata.cpp [hunk 1]
```diff
@@ -14,8 +14,9 @@ void ItemStackMetadata::serialize(std::ostream &os) const
 	std::ostringstream os2;
 	os2 << DESERIALIZE_START;
 	for (const auto &stringvar : m_stringvars) {
-		os2 << stringvar.first << DESERIALIZE_KV_DELIM
-		    << stringvar.second << DESERIALIZE_PAIR_DELIM;
+		if (stringvar.first != "" || stringvar.second != "")
```
Comment (nerzhul) [inline src/itemstackmetadata.cpp]: please use empty, as clang-tidy reports, it's faster than comparing empty std::string

if (!stringvar.first.empty() ...)

**Comment by paramat** [comment]
Added WIP to warn against merge.

**Comment by SmallJoker** [comment]
How does this need testing? The Metadata `getString` function will return an empty string if the key wasn't found. So this will be exactly the same as if the empty string was found in the map.

**Comment by rubenwardy** [comment]
checked, should be fine

**Comment by paramat** [comment]
557bbc670451f82c384032383df80a1a92ffe724
